adr/2025_03_10-suggestions/README.md

```diff
@@ -0,0 +1,172 @@
+
+This would be the most straightforward way to capture changes across versions of a document. The question then becomes how to display those changes within a prosemirror-based editor, which sort of depends on the approach taken for suggestions.
+
+### How should suggestions be stored
+
+There are a few different mental models we can use to represent or derive these changes:
+
+- [_Branched document histories_, and diffs between them](./suggestions-as-history.md)
+- [Stored _patches_ and later applying that patch to a document](./suggestions-as-patches.md)
+- [New _content types_ which represent the changes within the document content](./suggestions-as-content.md)
+
+### AttributionManager
+
+Currently, Y.js has been using a `PermanentUserData` class to keep track and store the `clientID`s that a specific user has used previously. Kevin has some ideas on an improved system for this, which he can elaborate on in detail, later. This will be crucial to how we can attribute users in prior versions of the document.
+
+If I were to share my wish-list for features this should have, it would be able to answer questions like:
+
+- Given a `clientID`, what `userID` made this change?
+- At what time was this `clientID` first used?
+  - This gives approximate times for attributing content to a time, without having to store an additional timestamp per change).
+  - Most people don't need the specific time, they just want to know relative time "Is this days, months or years old?"
+  - Auditing should be done at a different level, and not in a user writable data structure like a YDoc
+
+I'd like to have Kevin give more details on the sort of features/properties we can expect from this. It will be a crucial feature to the implementation, and likely to be worked on first since it will be foundational to how the versioning works.
+
+## Summary
+
+This document was purposefully written from the perspective that Y.js is one of many document storage and synchronization mechanisms. And, we would like to keep our editor de-coupled from these specific implementation details as much as possible to encourage code re-use between different implementations.
+
+## Appendix
+
+This is a collection of scattered thoughts on the _diff_-erent aspects of diffing  (sorry, had to), and their complications.
+
+### Diffing HTML
```

Comment:
This section implies there's something related to "HTML" that makes it difficult, I don't think this is the case. I would rename this section "Keeping user intent / operations intact" and describe it from a user instead of a technical perspective if that makes sense